src/components/app.tsx

```diff
@@ -5,13 +5,41 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import { Dispatch } from 'redux';
+import { StateWithHistory } from 'redux-undo';
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+import { datasetUrlLoad } from '../actions';
+import { StateBase } from '../models/index';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
 
-class AppBase extends React.PureComponent<any, {}> {
+export type VoyagerConfig = Object;
+export type VoyagerData = Data;
+
+interface Props extends React.Props<AppBase> {
+  config?: VoyagerConfig;
+  data?: Data;
+  dispatch: Dispatch<StateWithHistory<Readonly<StateBase>>>;
+}
+
+class AppBase extends React.PureComponent<Props, {}> {
+
+  constructor(props: any) {
```

Comment:
Also I've left the props optional for now as the app does not depend on them being passed in. As the config and other elements get fleshed out we can make them required and pass them in explicitly in all cases.